pkg/listen/listen.go

```diff
@@ -17,77 +17,111 @@ package listen
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"net/url"
 	"regexp"
+	"strings"
 
-	"github.com/hookdeck/hookdeck-cli/pkg/ansi"
 	"github.com/hookdeck/hookdeck-cli/pkg/config"
 	"github.com/hookdeck/hookdeck-cli/pkg/login"
 	"github.com/hookdeck/hookdeck-cli/pkg/proxy"
+	hookdecksdk "github.com/hookdeck/hookdeck-go-sdk"
 	log "github.com/sirupsen/logrus"
 )
 
 type Flags struct {
-	NoWSS bool
+	NoWSS   bool
+	CliPath string
 }
 
 // listenCmd represents the listen command
-func Listen(URL *url.URL, source_alias string, connectionQuery string, flags Flags, config *config.Config) error {
+func Listen(URL *url.URL, sourceQuery string, connectionFilterString string, flags Flags, config *config.Config) error {
 	var err error
-	var guest_url string
+	var guestURL string
+
+	sourceAliases, err := parseSourceQuery(sourceQuery)
+	if err != nil {
+		return err
+	}
+
+	isMultiSource := len(sourceAliases) > 1 || (len(sourceAliases) == 1 && sourceAliases[0] == "*")
+
+	if flags.CliPath != "" {
+		if isMultiSource {
+			return errors.New("Can only set a CLI path when listening to a single source")
+		}
+
+		_, err = isPath(flags.CliPath)
+		if err != nil {
+			return errors.New("The CLI path must be in a valid format")
+		}
+	}
 
 	if config.Profile.APIKey == "" {
-		guest_url, err = login.GuestLogin(config)
-		if guest_url == "" {
+		guestURL, err = login.GuestLogin(config)
+		if guestURL == "" {
 			return err
 		}
 	}
 
 	sdkClient := config.GetClient()
 
-	source, err := getSource(sdkClient, source_alias)
+	// Prepare data
+
+	sources, err := getSources(sdkClient, sourceAliases)
 	if err != nil {
 		return err
 	}
 
-	connections, err := getConnections(sdkClient, source, connectionQuery)
+	connections, err := getConnections(sdkClient, sources, connectionFilterString, isMultiSource, flags.CliPath)
 	if err != nil {
 		return err
 	}
 
-	fmt.Println()
-	fmt.Println(ansi.Bold("Dashboard"))
-	if guest_url != "" {
-		fmt.Println("👤 Console URL: " + guest_url)
-		fmt.Println("Sign up in the Console to make your webhook URL permanent.")
-		fmt.Println()
-	} else {
-		var url = config.DashboardBaseURL
-		if config.Profile.TeamID != "" {
-			url += "?team_id=" + config.Profile.TeamID
-		}
-		if config.Profile.TeamMode == "console" {
-			url = config.ConsoleBaseURL + "?source_id=" + source.Id
-		}
-		fmt.Println("👉 Inspect and replay events: " + url)
-		fmt.Println()
+	if len(flags.CliPath) != 0 && len(connections) > 1 {
+		return errors.New(fmt.Errorf(`Multiple CLI destinations found. Cannot set the CLI path on multiple destinations.
+Specify a single destination to update the CLI path. For example, pass a connection name:
+			
+  hookdeck listen %s %s %s --cli-path %s`, URL.String(), sourceQuery, "connection-name", flags.CliPath).Error())
```

Comment:
![CleanShot 2024-07-29 at 19 47 44](https://github.com/user-attachments/assets/df8d877d-a48f-47fe-b8e6-a36a25665e6d)

I run into an edge case here where you're assuming that the user passes a source name to the command (sourceQuery). However, this can also happen when the user doesn't pass anything, so the message doesn't quite work anymore.

Reply:
Ah, ok, so the message needs to take an empty source name being passed into account. I'll rectify.

Reply:
Fixed in e18b4d0036fb637d797ca026a0153ddbed2d797e